core/src/main/java/org/mule/runtime/core/internal/construct/FlowBackPressureException.java

```diff
@@ -44,17 +43,27 @@ public FlowBackPressureException(Component flow, BackPressureReason reason, Thro
     this.flow = flow;
   }
 
+  /**
+   * @return An instance of the right type of {@link FlowBackPressureException} according to the given {@link BackPressureReason}.
+   */
+  public static FlowBackPressureException createFlowBackPressureException(Component flow, BackPressureReason reason) {
+    return createFlowBackPressureException(flow, reason, null);
```

Comment:
Note that I'm passing `null` as cause here in order to reuse the other pre-existing version which expects the three parameters. One possible concern is that by doing this we are never using the two parameters constructor from `FlowBackPressureException` and its subclasses. But in the end, they all end up calling the `MuleException` constructor passing a `null` value when there is no cause.

If I wanted to avoid this I would either have to duplicate the switch-case, or make the code a lot less readable.